### DISCUSSION TITLE
"npm run dev" not working

### DISCUSSION THREAD

**Discussion by olajosh84** [vercel/next.js#57951] [Help] [answered]
### Summary

After installing next.js 14 on my PC(windows 10) and running "npm run dev", i get the following error

 ⚠ Attempted to load @next/swc-win32-ia32-msvc, but an error occurred: A dynamic link library (DLL) initialization routine failed.
\\?\C:\nextjs boilerplate\nextjs-boilerplate\node_modules\@next\swc-win32-ia32-msvc\next-swc.win32-ia32-msvc.node
 ⨯ Failed to load SWC binary for win32/ia32, see more info here: https://nextjs.org/docs/messages/failed-loading-swc.

Please what do i need to do to fix this issue?

**Comment by icyJoseph** [comment, marked as answer]
One thing that's often causing issues is, having Node.js 32 bits on a 64 bits OS. 

Run this command:

```console
node -p 'process.arch'
```

Whatever it prints, should match your OS' architecture.

**Reply by m07mmad-nasr** [reply]
True

**Reply by olajosh84** [reply]
I apparently downloaded the wrong node.js set up. I downloaded x86 instead of x64

**Reply by imran-khani** [reply]
Thank you. I just noticed I download x86 over x64

**Reply by ghost** [reply]
For some, reason when I ran command It's simply printing back the `process.arch` I think it's due to the single quotes If anyone else faces the same problem, I suggest running the command with double quotes instead.

```
node -p "process.arch"
```

Thank you.

**Reply by pcodesdev** [reply]
Thank you soo much this has saved me big time. I have been debugging my app for the last one hour but this solution has work. I was using node version 32 on an x64 machine. I have uninstalled the x86 version and installed the x64 version and everything is now working like a magic.
![nextjs](https://github.com/vercel/next.js/assets/44283776/0afb9b8d-98d3-47a9-a47b-eca2446a9508)

**Reply by Balamathias** [reply]
Oh certainly ! Installing the right one for my architecture fixed the issue.

I had spent six hours debugging until I stormed on this!

**Reply by Baryonae** [reply]
thank you so much!
i was using the X86 version indeed... 
i was stuck for so many days for this lol

**Reply by MuhammmadRizo** [reply]
Thank's it's TRUE

**Reply by Lakhan3011** [reply]
I download the x86 instead of x64 , thanks for this. I spent almost one night in figuring and debugging what's the problem

**Comment by Emmaxfans** [comment]
where am i going to run this command cos it doesn't seem to work here

**Reply by EmmanuelOmoiya** [reply]
You run it in your terminal / command prompt (windows).

**Reply by shadabahmed-code** [reply]
this commend not working in my pc

**Comment by jaber1988** [comment]
you haveHey there!

I understand you're encountering problems with Node.js on your 64-bit system. It seems you might have a 32-bit version of Node.js installed, which can cause compatibility issues.

Here's how to resolve this:

1. Uninstall the 32-bit Node.js:

Open the Control Panel.
Go to Programs and Features (or Add or Remove Programs, depending on your Windows version).
Locate Node.js in the program list.
Right-click on it and select Uninstall.
Follow the on-screen prompts to complete the uninstallation.
2. Download and Install the 64-bit Node.js:

Head over to the official Node.js download page: https://nodejs.org/en/download
Under the LTS (Long-Term Support) section, download the Windows Installer (.msi) corresponding to the x64 architecture.
Double-click the downloaded installer and follow the on-screen instructions to complete the installation.
3. Verify the Installation:

Open a command prompt window (Start menu, search for cmd, press Enter).
Type node -v and press Enter. This should display the newly installed 64-bit Node.js version.
4. Optional: Restart Your Computer (Recommended):

While not strictly necessary, restarting your computer ensures any system-wide environment variables are updated to reflect the new Node.js installation.
Additional Tips:

Package Manager Compatibility: If you're using a package manager like npm or yarn, you may need to re-install your global packages after switching to the 64-bit version.
By following these steps, you should have Node.js 64-bit successfully installed and running on your system. If you encounter any further issues, feel free to provide more details, and the community can assist you further.
